osf/models/mixins.py

```diff
@@ -1991,8 +1992,9 @@ def confirm_ham(self, save=False):
         super().confirm_ham()
 
         if self.logs.filter(action__in=[self.log_class.FLAG_SPAM, self.log_class.CONFIRM_SPAM]):
+            spam_log = self.logs.filter(action__in=[self.log_class.FLAG_SPAM, self.log_class.CONFIRM_SPAM]).latest()
             # ensures only 'accepted' status preprints/any nodes get made public
-            if self.type == 'osf.node' or (self.type == 'osf.preprint' and self.machine_state == DefaultStates.ACCEPTED.value):
+            if spam_log.params.get('was_public', False) and self.type == 'osf.node' or (self.type == 'osf.preprint' and self.machine_state == DefaultStates.ACCEPTED.value):
```

Comment:
Should there be parentheses around `self.type == 'osf.node' or (self.type == 'osf.preprint' and self.machine_state == DefaultStates.ACCEPTED.value)`?